src/qibolab/serialize.py

```diff
@@ -205,6 +205,35 @@ def dump_runcard(platform: Platform, path: Path):
         platform.qubits, platform.couplers
     )
 
-    path.write_text(
+    (path / RUNCARD).write_text(
         yaml.dump(settings, sort_keys=False, indent=4, default_flow_style=None)
     )
+
+
+def dump_kernels(platform: Platform, path: Path):
+    """Creates Kernels instance from platform and dumps as npz.
+
+    Args:
+        platform (qibolab.platform.Platform): The platform to be serialized.
+        path (pathlib.Path): Path that the kernels file will be saved.
+    """
+
+    # create and dump kernels
+    kernels = Kernels()
+    for qubit in platform.qubits.values():
+        if qubit.kernel is not None:
+            kernels[qubit.name] = qubit.kernel
+
+    kernels.dump(path / KERNELS)
```

Comment:
I think this will not work. `KERNELS` is appended to the `path` twice, once here and once in https://github.com/qiboteam/qibolab/blob/5e46e6f43bc6a6a0920d0ee669a48f0e26c2a4e9/src/qibolab/kernels.py#L37

We should test the PRs better before considering to merge. Particularly the new `dump_kernels` and `dump_platform` methods, I believe are not captured by the tests, while they can be easily tested without access to instruments/QPUs.

Reply:
Yes, I also found the same error while trying to run tests in qibocal. 
I will try to add tests to cover `dump_platform` and `dump_kernels`.

Reply:
Tests added in https://github.com/qiboteam/qibolab/pull/758/commits/8f49f89ec3706b9b21f23441d96813a4eb375bb8